### DISCUSSION TITLE
I2C0 works but I2C1 doesn't

### DISCUSSION THREAD

**Discussion by w9ran** [earlephilhower/arduino-pico#2672] [Q&A] [answered]
Here is a simple test program to check out .96" oled displays.   If I connect the display to GPIO 16/17 and enable I2C0,  it works.   But if I move the display to GPIO 18/19 and enable I2C1,  the display remains blank.   I've tried other legal I2C1 pin combinations with the same result - I2C0 always works and I2C1 never works.   I can't see where I'm doing anything wrong so really would appreciate any suggestions, thanks.

#include <SPI.h>
#include <Wire.h>
#include <Adafruit_GFX.h>
#include <Adafruit_SSD1306.h>

#define SCREEN_WIDTH 128 // OLED display width, in pixels
#define SCREEN_HEIGHT 64 // OLED display height, in pixels
#define SCREEN_ADDRESS 0x3C ///< See datasheet for Address; 0x3C 

// I2C pins are defined in Raspberry Pi Pico pins-arduino.h file for mbed

Adafruit_SSD1306 display(SCREEN_WIDTH, SCREEN_HEIGHT, &Wire, OLED_RESET);

void setup() {
 // i2c works
// Wire.setSDA(16);
// Wire.setSCL(17);
// Wire.begin();

// i2c1 does not work
 Wire1.setSDA(18);
 Wire1.setSCL(19);
 Wire1.begin();
 display.begin(SSD1306_SWITCHCAPVCC, SCREEN_ADDRESS);  // initialize display
  // Clear the buffer
  display.clearDisplay();
  display.setTextSize(2);             // Normal 1:1 pixel scale
  display.setTextColor(SSD1306_WHITE);        // Draw white text
  display.setCursor(20,30);             
  display.println("I2C TEST");
  display.display();
 }
void loop() {
}

**Comment by earlephilhower** [comment]
I'm not seeing your code tell the display library to use `Wire1` and not `Wire`, so check if there's a constructor or `begin` method to allow that.

**Reply by Andy2No** [reply, marked as answer]
Yep.  It looks like this line would need to be changed, to use Wire1 instead:

```
Adafruit_SSD1306 display(SCREEN_WIDTH, SCREEN_HEIGHT, &Wire, OLED_RESET);
```

Try changing  **&Wire** to **&Wire1**.

**Comment by w9ran** [comment]
Thank you!   That never occurred to me, but now I know the proper usage.
